client/src/app/components/d-flow-profile-edit/d-flow-profile-edit.component.ts

```diff
@@ -0,0 +1,160 @@
+    info_infuse_stop: 'Infuse will move on to the pour stage when any one of these settings are reached. \n For best consistancy Damian recommends using weight. In cases where a scale is not available,      you could use volume, however for those situations and where you you may want to end infuse before first drops in the cup, using time is likely the better option. \n\nA longer infusion increases body, but it also reduces puck resistance during the pour. Which means we would need to grind finer to maintain the same pour pressure/flow rates. Finer grinds also reduce body and can increase bitterness, so the aim is to find a ballence for your prefered taste. \n\n A longer infusion also means we have a higher pecentage of the shot being at 3 bar.\n 0.2g to 4g is a good starting range to experiment with.',
+    info_pour_temp: 'Higher temperature helps extract the soluble compounds, however it also increases harsher tasts that may exist too. \n\n Higher temperatures usually results in lower body with more pronounced acidity, wine, fruity, vegetal flavours. \nLower temperatures often work better with slower extraction rates.',
+    info_pour_limits: 'D-Flow uses flow with pressure limits, allows for a greater grind range, giving better control and resulting in less wasted shots. \n Increasing pressure will shift taste from wine like to a more syrupy texture, it also shifts tastes from clear delicate flavours to more muddled flavours.\n\nFaster flows can increase clarity in ligher roasts and may help reduce channeling. \n Slower flow can help increase intensity and body in darker than light roasts.',
+    info_pour_stop: 'The extraction ratio (Dose : Extraction) is shown above the setting dial.\n Increasing the extraction ratio will shift the taste from \n\n Sour  >  Sweet  >  Bitter \n\nThe ideal extraction ratio can vary between beans, water alkalinity, puck prep methods and how evenly the pack is extracted.\n You should adjusted this setting for your taste.',
+    info_d_flow: 'The D-Flow editor was developed by Damian. It is available as a plug-in for the de1app. The original code can be found on Github: https://github.com/Damian-AU/D_Flow_Espresso_Profile'
+  };
+
+  @ViewChild('chart')
+  chart!: ProfileChartComponent;
+
+  @Input()
+  profile: Profile | undefined;
+
+  @Output()
+  changed: EventEmitter<Profile> = new EventEmitter();
+
+  _profileService = inject(ProfileServiceService);
+
+  updateExtractionRatio() {
+    if (this.profile?.target_weight && this.profile?.target_weight != 0) {
+      this.extractionRatio = this.profile?.target_weight / this.dose;
+    }
+    else if (this.profile?.target_volume && this.profile?.target_volume != 0) {
+      this.extractionRatio = this.profile?.target_volume / this.dose;
+    }
+  }
+
+  save() {
+    this._profileService.updateProfile(this.profile!);
+  }
+
+  resetToDefaults() {
+    if (this.profile) {
+      this.profile.author = 'Damian';

```

Comment:
Kann man hier nicht einfach eines der in assets liegenden Profile laden? Es gibt eine loadById Funktion dafür die ein Profil generiert.

Reply:
So wirds halt im Original Plugin gemacht. Das andere habe ich mir auch überlegt - woher weiss ich die ID?

Reply:
getAssetProfileById(id: string)
Die ID's stehen hier:
allProfiles: string[] = [
    "Blooming espresso.json",
    "Classic Italian espresso.json",
    "Cleaning__Weber_Spring_Clean.json",
    "Cleaning_forward_flush_x5.json",
    "Cremina.json",
    "D-Flow____default.json",
    "Damian's LM Leva.json",
    "Damian's LRv2.json",
    "Damian's LRv3.json",
    "Default.json",
    "Easy_blooming_pressure_dynamic_decline.json",
    "Extractamundo_Dos.json",
    "Filter_20.json",
    ...
    
 Vermutlich wäre d-flow____default hier am besten?
 Zur Not einfach ein neues einfügen und das dann als Reset nehmen mit den Werten die du gerne setzen willst.